### ISSUE TITLE
Make sure validation can be done using several batches!

### ISSUE THREAD

**Issue by johann-petrak** [GateNLP/gate-lf-pytorch-json#46]
Make sure we can use the same batchsize as for training for evaluating on the validation set,
refactor things so that this gets merged with the code currently only used with elmo models 
(and using fixed batch sizes!)
